src/MongoDB.Driver/EncryptOptions.cs

```diff
@@ -68,5 +69,14 @@ public class EncryptOptions
         /// The key identifier.
         /// </value>
         public byte[] KeyId => _keyId;
+
+        // private methods
+        private static void EnsureThatOptionsAreValid(byte[] keyId, string keyAltName)
+        {
+            Ensure.That(
```

Comment:
This method is required since we need to decide which method we will use to create `ExplicitEncryptionContext`. In the current implementation, we cannot push all provided options and rely on the server